cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -274,22 +277,29 @@ struct preprocessed_table {
    * `null_order::BEFORE` for all columns.
    * @param stream The stream to launch kernels and h->d copies on while preprocessing.
    */
-  preprocessed_table(table_view const& table,
-                     host_span<order const> column_order,
-                     host_span<null_order const> null_precedence,
-                     rmm::cuda_stream_view stream);
-
-  // TODO: Should we add a static create method that returns a shared_ptr?
+  static std::shared_ptr<preprocessed_table> create(table_view const& table,
+                                                    host_span<order const> column_order,
+                                                    host_span<null_order const> null_precedence,
+                                                    rmm::cuda_stream_view stream);
```

Comment:
Do we really want to force people to always construct as a `shared_ptr`?

I suppose if someone is going through the trouble of constructing a `preprocessed_table` themselves, then that means they are intending to use it in more than one comparator, in which case it will already need to be a `shared_ptr`.

Reply:
exactly. Why not return a shared_ptr if everything that can accept it needs a shared_ptr.